src/v/crypto/tests/ossl_helpers.bzl

```diff
@@ -0,0 +1,41 @@
+"""
+This module contains helpers for working with ossl modules
+"""
+
+def _copy_gen_file(ctx):
+    target = ctx.attr.source
+
+    # NOTE(oren): pretty sure it's always just the one
+    gen_dir = target[OutputGroupInfo].gen_dir.to_list()[0]
+    dest = ctx.attr.dest
+    if dest == "":
+        dest = ctx.attr.path
+    out = ctx.actions.declare_file(dest)
+    ctx.actions.run_shell(
+        outputs = [out],
+        inputs = [gen_dir],
+        arguments = ["{}/{}".format(gen_dir.path, ctx.attr.path), out.path],
+        command = "cp $1 $2",
+    )
+    return [
+        DefaultInfo(files = depset([out])),
+    ]
+
+copy_gen_file = rule(
```

Comment:
I'm 99.9% sure this can be replaced with https://github.com/bazelbuild/bazel-skylib/blob/main/docs/select_file_doc.md

We already have bazel skylib included as a dependency.

Reply:
Why is this stuff so hard to discover 😭

Reply:
So this works for the fips module, but doesn't seem to for `fipsmodule.cnf`. I think this is because of the openssl.conf file itself - we need to populate it with an absolute or relative (to the test's cwd at runtime) path to the fipsmodule.cnf. 

So in practice we need `fipsmodule.cnf` (as generated by the openssl-fips build) to appear in a well (ideally statically) known location in each test's runfiles. I don't see a straightforward way to do this with existing rules. Any ideas?

Reply:
Maybe the problem is more that `fipsmodule.cnf` doesn't appear directly in `@openssl-fips` files list. That seems to be a limitation of `configure_make` - all you get is `out_data_dirs`... 🤷

Reply:
Yeah annoyingly the out data dirs are declared as [tree artifacts](https://github.com/bazel-contrib/rules_foreign_cc/blob/bcd0ec409f383a26602d266c5f50523b28ee88a9/foreign_cc/private/framework.bzl#L819), which are a bit clunky. I will look into this a bit.

Reply:
Yeah feels like we're stuck w/ some version of manipulating artifacts directly, but I'm just shooting the breeze. Thanks!

Reply:
it's gone 🥳